agent/functions/put_file_test.go

```diff
@@ -0,0 +1,45 @@
+package functions_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github/clover0/github-issue-agent/functions"
+)
+
+func TestPutFile(t *testing.T) {
+	t.Run("successfully writes content to file", func(t *testing.T) {
+		outputPath := "testdata/testfile.txt"
+		content := "Hello, World!"
+
+		defer os.Remove(outputPath)
+
+		file, err := functions.PutFile(functions.PutFileInput{
+			OutputPath:  outputPath,
+			ContentText: content,
+		})
+
+		require.NoError(t, err)
+		assert.Equal(t, outputPath, file.Path)
+		assert.Equal(t, content+"\n", file.Content)
+
+		actualContent, err := os.ReadFile(outputPath)
+		require.NoError(t, err)
+		assert.Equal(t, content+"\n", string(actualContent))
+	})
+
+	t.Run("fails with invalid path", func(t *testing.T) {
+		outputPath := "invalid_path/testfile.txt"
+		content := "Hello, World!"
+
+		_, err := functions.PutFile(functions.PutFileInput{
+			OutputPath:  outputPath,
+			ContentText: content,
+		})
+
```

Comment:
from SecurityExpert
テストコードでは、外部からの入力を受け取る部分はありませんが、ファイルパスの操作が含まれています。ファイルパスの操作においては、ディレクトリトラバーサル攻撃を防ぐために、パスの正規化や検証を行うことが推奨されます。

```suggestion
outputPath := filepath.Clean("testdata/testfile.txt")
```